src/components/AbstractModal/AbstractModal.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import Modal from 'react-modal';
+
+const AbstractModal = ({ modalId, className, required, modalIsOpen, children }) => {
+    return (
+        <Modal 
+            id={ modalId } 
+            className={ className }
+            shouldCloseOnOverlayClick={ !required }
+            shouldCloseOnEsc={ !required }
+            isOpen={ modalIsOpen }
```

Comment:
fix: I noticed a recurring error in the dev console that appElement was not assigned or something like that. It has to do with accessibility. The app behind our modal has to be hidden from screen readers. 
This bit of code will fix it:

`appElement={document.getElementById('root') || undefined}`

just add it as another prop of the `<Modal>` component.

Reply:
done